compile.jai

```diff
@@ -147,8 +147,9 @@ Compiler :: struct {
 // The reversed flag is also recorded in the returned program.
 compile :: (re: *Regexp_Node, pool: *Regexp_Pool, reversed := false, max_mem := 0) -> *Prog {
 	new_context := context;
-	new_context.allocator = pool_allocator;
-	new_context.allocator_data = pool;
+
+	// FIXME: For some reason, uncommenting this line causes everything to crash
+	//new_context.allocator = pool.allocator;
```

Comment:
It crashes because you set pool.allocator.data to a pointer to pool early on. But then `Pool` gets copied around, so `allocator.data` now points to invalid memory on the stack somewhere.

Reply:
(The old code had the same problem with `Regexp_Pool.bucket.allocator_data`, but it didn’t manifest because the bucket only allocates while the original pointer is still valid.)

Reply:
Ah! Excellent! I'm glad you found that, it was super confusing.